Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -323,12 +328,14 @@ - (instancetype) initWithConfig: (CBLReplicatorConfiguration*)config
 // TODO: Remove https://issues.couchbase.com/browse/CBL-3206
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-        _database = config.database;
-        _documentIDs = config.documentIDs;
-        _channels = config.channels;
-        _pushFilter = config.pushFilter;
-        _pullFilter = config.pullFilter;
-        _conflictResolver = config.conflictResolver;
+        [self initCollectionConfigs];
```

Comment:
I'm not sure if this is required. I think that it needs is to copy _collectionConfigs from the other config and verify if the collections are valid && database is not nil.

Reply:
yes, right. But why do we need to check database. I guess, I can verify each collection is valid, which will internally check the db open or not for that collection.